--- mgetty-1.1.35/callback/Makefile.noroot	2000-08-14 21:33:08.000000000 +0200
+++ mgetty-1.1.35/callback/Makefile	2007-04-23 14:47:42.000000000 +0200
@@ -40,7 +40,7 @@
 # install programs
 #
 install: all
-	$(INSTALL) -s -m 700 -o root callback $(SBINDIR)
-	$(INSTALL) -s -m 4711 -o root ct $(BINDIR)
+	$(INSTALL) -s -m 700 callback $(SBINDIR)
+	$(INSTALL) -s -m 4711 ct $(BINDIR)
 
 # source modules
--- mgetty-1.1.35/contrib/next-login/Makefile.noroot	1996-05-26 18:23:06.000000000 +0200
+++ mgetty-1.1.35/contrib/next-login/Makefile	2007-04-23 14:47:42.000000000 +0200
@@ -42,7 +42,7 @@
 	mkdep -p ${CFLAGS} ${SRCS}
 
 install:
-	install -s -o root -g bin -m 4755 login ${DESTDIR}/bin/modem-login
+	install -s -m 4755 login ${DESTDIR}/bin/modem-login
 
 lint: ${SRCS}
 	lint ${CFLAGS} ${SRCS}
--- mgetty-1.1.35/fax/Makefile.noroot	2005-02-16 16:54:46.000000000 +0100
+++ mgetty-1.1.35/fax/Makefile	2007-04-23 14:47:42.000000000 +0200
@@ -61,7 +61,6 @@
 	done
 	$(INSTALL) -m 755 faxrunqd $(SBINDIR)
 	$(INSTALL) faxq-helper $(LIBDIR)
-	chown $(FAX_OUT_USER) $(LIBDIR)/faxq-helper
 	chmod 4711 $(LIBDIR)/faxq-helper
 #
 # data files
--- mgetty-1.1.35/frontends/www/Makefile.noroot	2004-11-24 14:46:17.000000000 +0100
+++ mgetty-1.1.35/frontends/www/Makefile	2007-04-23 14:47:42.000000000 +0200
@@ -95,7 +95,7 @@
 		    $(INSTALL) -m 644 $$i $(HTMLDIR) ; \
 	    done
 	test -f $(CONFDIR)/wwwgui.conf || \
-		$(INSTALL) -o root -m 644 wwwgui.conf $(CONFDIR)/
+		$(INSTALL) -m 644 wwwgui.conf $(CONFDIR)/
 	$(INSTALL) -m 755 simplequant.pl $(LIBDIR)/
 	$(INSTALL) -m 755 wwwsub.pl $(LIBDIR)/
 
--- mgetty-1.1.35/Makefile.noroot	2007-04-23 14:47:42.000000000 +0200
+++ mgetty-1.1.35/Makefile	2007-04-23 14:48:00.000000000 +0200
@@ -164,7 +164,7 @@
 #
 # if your systems doesn't have one, use the shell script that I provide
 # in "inst.sh" (taken from X11R5). Needed on IRIX5.2
-INSTALL=install -c -o root -g root
+INSTALL=install -c
 #INSTALL=install -c -o root -g wheel		# NeXT/BSD
 #INSTALL=/usr/ucb/install -c -o bin -g bin	# AIX, Solaris 2.x
 #INSTALL=installbsd -c -o bin -g bin		# OSF/1, AIX 4.1, 4.2
@@ -602,15 +602,15 @@
 	test -d $(CONFDIR) || \
 		( ./mkidirs $(CONFDIR) && chmod 755 $(CONFDIR))
 	test -f $(CONFDIR)/login.config || \
-		$(INSTALL) -o root -m 600 login.config $(CONFDIR)/
+		$(INSTALL) -m 600 login.config $(CONFDIR)/
 	test -f $(CONFDIR)/mgetty.config || \
-		$(INSTALL) -o root -m 600 mgetty.config $(CONFDIR)/
+		$(INSTALL) -m 600 mgetty.config $(CONFDIR)/
 	test -f $(CONFDIR)/sendfax.config || \
-		$(INSTALL) -o root -m 644 sendfax.config $(CONFDIR)/
+		$(INSTALL) -m 644 sendfax.config $(CONFDIR)/
 	test -f $(CONFDIR)/dialin.config || \
-		$(INSTALL) -o root -m 600 dialin.config $(CONFDIR)/
+		$(INSTALL) -m 600 dialin.config $(CONFDIR)/
 	test -f $(CONFDIR)/faxrunq.config || \
-		$(INSTALL) -o root -m 644 faxrunq.config $(CONFDIR)/
+		$(INSTALL) -m 644 faxrunq.config $(CONFDIR)/
 #
 # test for outdated stuff
 #
@@ -628,13 +628,11 @@
 		( mkdir $(spool) && chmod 755 $(spool) )
 	test -d $(FAX_SPOOL) || \
 		( mkdir $(FAX_SPOOL) && \
-		  chown $(FAX_OUT_USER) $(FAX_SPOOL) && \
 		  chmod 755 $(FAX_SPOOL) )
 	test -d $(FAX_SPOOL_IN) || \
 		( mkdir $(FAX_SPOOL_IN) && chmod 755 $(FAX_SPOOL_IN) )
 	test -d $(FAX_SPOOL_OUT) || \
 		  mkdir $(FAX_SPOOL_OUT)
-	chown $(FAX_OUT_USER) $(FAX_SPOOL_OUT)
 	chmod 755 $(FAX_SPOOL_OUT)
 #
 # g3 tool programs
