diff --git a/Makefile b/Makefile
index fb5d4af..2c3f85c 100644
--- a/Makefile
+++ b/Makefile
@@ -171,7 +171,8 @@ LIBS=-lutil					# FreeBSD or Linux/GNU libc2
 #
 # if your systems doesn't have one, use the shell script that I provide
 # in "inst.sh" (taken from X11R5). Needed on IRIX5.2
-INSTALL=install -c -o bin -g bin
+INSTALL=install -c
+#INSTALL=install -c -o bin -g bin
 #INSTALL=install -c -o root -g wheel		# NeXT/BSD
 #INSTALL=/usr/ucb/install -c -o bin -g bin	# AIX, Solaris 2.x
 #INSTALL=installbsd -c -o bin -g bin		# OSF/1, AIX 4.1, 4.2
