diff --git a/Makefile b/Makefile
index c8ed7c8..4acb69b 100644
--- a/Makefile
+++ b/Makefile
@@ -109,7 +109,7 @@ HOSTCC=$(CC)
 #	    USTAT	  - ustat(), no statfs etc.
 #
 #CFLAGS=-Wall -O2 -pipe -DSECUREWARE -DUSE_POLL
-CFLAGS=-Wall -O2 -pipe $(RPM_OPT_FLAGS)
+CFLAGS=-Wall -O2 -pipe -fpie
 #CFLAGS=-O -DSVR4
 #CFLAGS=-O -DSVR4 -DSVR42
 #CFLAGS=-O -DUSE_POLL
@@ -150,7 +150,7 @@ CFLAGS=-Wall -O2 -pipe $(RPM_OPT_FLAGS)
 # 	"utmp.o: unresolved symbol _login"
 # For Linux, add "-lutil" if the linker complains about "updwtmp".
 #
-LDFLAGS=$(RPM_LD_FLAGS)
+LDFLAGS=-z now -pie 
 #LIBS=
 #LIBS=-lprot -lsocket				# SCO Unix
 #LIBS=-lsocket
@@ -618,20 +618,19 @@ install.bin: mgetty sendfax newslock \
 	test -d $(CONFDIR) || \
 		( ./mkidirs $(CONFDIR) && chmod 755 $(CONFDIR))
 	test -f $(CONFDIR)/login.config || \
-		( $(INSTALL) -m 600 login.config $(CONFDIR)/ && \
-		  $(CHOWN) root $(CONFDIR)/login.config )
+		( $(INSTALL) -m 600 login.config $(CONFDIR) )
 	test -f $(CONFDIR)/mgetty.config || \
-		( $(INSTALL) -m 600 mgetty.config $(CONFDIR)/ && \
-		  $(CHOWN) root $(CONFDIR)/mgetty.config )
+		( $(INSTALL) -m 600 mgetty.config $(CONFDIR) ) 
+#		  $(CHOWN) root $(CONFDIR)/mgetty.config )
 	test -f $(CONFDIR)/sendfax.config || \
-		( $(INSTALL) -m 644 sendfax.config $(CONFDIR)/ && \
-		  $(CHOWN) root $(CONFDIR)/sendfax.config )
+		( $(INSTALL) -m 644 sendfax.config $(CONFDIR) )
+#		  $(CHOWN) root $(CONFDIR)/sendfax.config )
 	test -f $(CONFDIR)/dialin.config || \
-		( $(INSTALL) -m 600 dialin.config $(CONFDIR)/ && \
-		  $(CHOWN) root $(CONFDIR)/dialin.config )
+		( $(INSTALL) -m 600 dialin.config $(CONFDIR) )
+#		  $(CHOWN) root $(CONFDIR)/dialin.config )
 	test -f $(CONFDIR)/faxrunq.config || \
-		( $(INSTALL) -m 644 faxrunq.config $(CONFDIR)/ && \
-		  $(CHOWN) root $(CONFDIR)/faxrunq.config )
+		( $(INSTALL) -m 644 faxrunq.config $(CONFDIR) )
+#		  $(CHOWN) root $(CONFDIR)/faxrunq.config )
 #
 # test for outdated stuff
 #
diff --git a/callback/Makefile b/callback/Makefile
index 236fe34..ec708ae 100644
--- a/callback/Makefile
+++ b/callback/Makefile
@@ -37,9 +37,9 @@ clean:
 # install programs
 #
 install: all
-	$(INSTALL_S) -m 700 callback $(SBINDIR)
+	$(INSTALL_S) -m 700 -o root callback $(SBINDIR)
 	$(CHOWN) root $(SBINDIR)/callback
-	$(INSTALL_S) -m 4711 ct $(BINDIR)
+	$(INSTALL_S) -m 4711 -o root ct $(BINDIR)
 	$(CHOWN) root $(BINDIR)/ct
 
 # source modules
diff --git a/frontends/X11/viewfax/Makefile b/frontends/X11/viewfax/Makefile
index abe3b18..fb30f54 100644
--- a/frontends/X11/viewfax/Makefile
+++ b/frontends/X11/viewfax/Makefile
@@ -38,20 +38,20 @@ OPT = -g -O2 -Wno-uninitialized -ansi -pedantic
 #LDFLAGS = $(OPT) $(RPM_LD_FLAGS)
 #LIBS = -lsocket
 # vanilla BSD
-CC = cc
-CFLAGS = -DBSD $(OPT) $(RPM_OPT_FLAGS) -DHELPFILE=$(HELP) -I/usr/local/include
-LDFLAGS = $(OPT) $(RPM_LD_FLAGS)
-LIBS =
+#CC = cc
+#CFLAGS = -DBSD $(OPT) $(RPM_OPT_FLAGS) -DHELPFILE=$(HELP) -I/usr/local/include
+#LDFLAGS = $(OPT) $(RPM_LD_FLAGS)
+#LIBS =
 # sun solaris2 with gcc on supersparc:
 #CC = gcc
 #CFLAGS = -Wall $(OPT) -DHELPFILE=$(HELP) -msupersparc -I/usr/openwin/include
 #LDFLAGS = $(OPT) -L/usr/openwin/lib -R/usr/openwin/lib
 #LIBS =
 # linux
-#CC = gcc
-#CFLAGS = $(OPT) $(RPM_OPT_FLAGS) -DHELPFILE=$(HELP)
-#LDFLAGS = $(OPT) $(RPM_LD_FLAGS) -L/usr/X11R6/lib
-#LIBS =
+CC = gcc
+CFLAGS = $(OPT) $(RPM_OPT_FLAGS) -DHELPFILE=$(HELP)
+LDFLAGS = $(OPT) $(RPM_LD_FLAGS) -L/usr/X11R6/lib
+LIBS =
 
 ####### End of configurable definitions #######
 OBJS = viewfax.o faxinput.o faxinit.o faxexpand.o
