--- mgetty-1.1.21/voice/Makefile.orig	Sat Nov 27 16:59:47 1999
+++ mgetty-1.1.21/voice/Makefile	Sat Nov 27 17:11:57 1999
@@ -93,7 +93,7 @@
 	cd $(MAN1DIR); \
 	for i in $(PVFTOOLS); \
 	do \
-	     $(RM) -f $$i.1; $(LN) -s pvf.1 $$i.1; \
+	     $(RM) -f $$i.1; $(LN) -s pvf.1.bz2 $$i.1.bz2; \
 	done
 #	[ -d $(VOICE_DIR) ] || mkdir $(VOICE_DIR)
 #	[ -d $(VOICE_DIR)/incoming ] || mkdir $(VOICE_DIR)/incoming
